taqueria-plugin-tezos-client/client.ts

```diff
@@ -1,17 +1,42 @@
-import { sendAsyncErr } from '@taqueria/node-sdk';
-import { IntersectionOpts as Opts } from './common';
-import simulate from './simulate';
-import typecheck from './typecheck';
+import { execCmd, getArch, getFlextesaImage, sendAsyncErr, sendErr, sendRes, sendWarn } from '@taqueria/node-sdk';
+import { writeFile } from 'fs/promises';
+import { ClientOpts as Opts } from './common';
 
-const client = (parsedArgs: Opts): Promise<void> => {
-	switch (parsedArgs.task) {
-		case 'typecheck':
-			return typecheck(parsedArgs);
-		case 'simulate':
-			return simulate(parsedArgs);
-		default:
-			return sendAsyncErr(`${parsedArgs.task} is not an understood task by the Tezos-client plugin`);
-	}
+const MAX_OUTPUT_LIMIT = 50000;
+
+const getArbitraryClientCmd = async (parsedArgs: Opts, cmd: string): Promise<string> => {
+	const projectDir = process.env.PROJECT_DIR ?? parsedArgs.projectDir;
+	if (!projectDir) throw `No project directory provided`;
+	const arch = await getArch();
+	const flextesaImage = await getFlextesaImage(arch);
+	const baseCmd =
+		`docker run --rm -v \"${projectDir}\":/project -w /project --platform ${arch} ${flextesaImage} octez-client`;
+	return `${baseCmd} ${cmd}`;
+};
+
+const runArbitraryClientCmd = (parsedArgs: Opts, cmd: string): Promise<string> =>
+	getArbitraryClientCmd(parsedArgs, cmd)
+		.then(execCmd)
+		.then(async ({ stdout, stderr }) => {
+			if (stderr.length > 0) sendWarn(stderr);
+			if (stdout.length >= MAX_OUTPUT_LIMIT) {
+				const outputFile = './octez-client.output';
```

Comment:
What might be best is to send a unique id (hash of microtime + request args might be sufficient) along with the request, and add that as a prefix/postfix to the filename used. This way users can invoke this ligo command with different purposes simultaneously without clobbering the result.

Reply:
At looking closer, although I think having a unique command invocation id passed along with plugin requests could be useful, rather than write to file in this case, we should avoid using execCmd in this case and write another method for the SDK which shares (or at least pipes) output to the main process with minimal buffering.